### ISSUE TITLE
Build Issue

### ISSUE THREAD

**Issue by hmoenck** [BioroboticsLab/pipeline#10]
I'm getting a build error on the master branch. Might also be related to https://github.com/BioroboticsLab/cmakeconfig
CMakeLists.txt snippet:

```
cmake_minimum_required(VERSION 2.6)
set(CPM_MODULE_NAME YourModuleNameHere)
set(CPM_LIB_TARGET_NAME ${CPM_MODULE_NAME})
    set(BIOROBOTICS_CMAKE_CONFIG_DIR "${CMAKE_CURRENT_BINARY_DIR}/cmakeconfig" CACHE TYPE STRING)
    find_package(Git)
    if(NOT GIT_FOUND)
      message(FATAL_ERROR "CPM requires Git.")
    endif()
    if (NOT EXISTS ${BIOROBOTICS_CMAKE_CONFIG_DIR}/CMakeLists.txt)
      message(STATUS "Cloning repo (https://github.com/BioroboticsLab/cmakeconfig.git)")
      execute_process(
        COMMAND "${GIT_EXECUTABLE}" clone https://github.com/BioroboticsLab/cmakeconfig.git
        ${BIOROBOTICS_CMAKE_CONFIG_DIR}
        RESULT_VARIABLE error_code
        OUTPUT_QUIET ERROR_QUIET)
      if(error_code)
          message(FATAL_ERROR "CMAKECONFIG failed to get the hash for HEAD")
      endif()
    endif()
    include(${BIOROBOTICS_CMAKE_CONFIG_DIR}/CMakeLists.txt)
CPM_AddModule("cmakeconfig"
    GIT_REPOSITORY "https://github.com/BioroboticsLab/cmakeconfig.git"
    GIT_TAG "master")
include_pipeline()
include_biotracker_core()
CPM_InitModule(${CPM_MODULE_NAME})
biorobotics_set_compiler_flags()
CPM_Finish()
```

Build Error:

```
[ 45%] Building CXX object cpm-bin/github_nebw_mxnetpredict_master/bin/CMakeFiles/github_nebw_mxnetpredict_master.dir/source/MXNetPredictor.cpp.o
<command-line>:0:0: error: "CPM_CMAKECONFIG_NS" redefined [-Werror]
<command-line>:0:0: note: this is the location of the previous definition
cc1plus: all warnings being treated as errors
cpm-bin/github_nebw_mxnetpredict_master/bin/CMakeFiles/github_nebw_mxnetpredict_master.dir/build.make:62: recipe for target 'cpm-bin/github_nebw_mxnetpredict_master/bin/CMakeFiles/github_nebw_mxnetpredict_master.dir/source/MXNetPredictor.cpp.o' failed
make[2]: *** [cpm-bin/github_nebw_mxnetpredict_master/bin/CMakeFiles/github_nebw_mxnetpredict_master.dir/source/MXNetPredictor.cpp.o] Error 1
CMakeFiles/Makefile2:269: recipe for target 'cpm-bin/github_nebw_mxnetpredict_master/bin/CMakeFiles/github_nebw_mxnetpredict_master.dir/all' failed
make[1]: *** [cpm-bin/github_nebw_mxnetpredict_master/bin/CMakeFiles/github_nebw_mxnetpredict_master.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by hmoenck** [comment]
Ok, I deleted the excess definition from:
cpm-bin/github_nebw_mxnetpredict_master/bin/CMakeFiles/github_nebw_mxnetpredict_master.dir/build.make
It fixed this particular issue, but many others followed:

```
<command-line>:0:0: error: "CPM_CMAKECONFIG_NS" redefined [-Werror]
<command-line>:0:0: note: this is the location of the previous definition
[ 49%] Building CXX object cpm-bin/github_bioroboticslab_pipeline_master/bin/CMakeFiles/github_bioroboticslab_pipeline_master.dir/pipeline/src/util/opencv_fill_poly/fill_poly_main.cpp.o
[ 49%] Building CXX object cpm-bin/github_bioroboticslab_pipeline_master/bin/CMakeFiles/github_bioroboticslab_pipeline_master.dir/pipeline/src/util/opencv_fill_poly/ring_iterator.cpp.o
[ 50%] Building CXX object cpm-bin/github_bioroboticslab_pipeline_master/bin/CMakeFiles/github_bioroboticslab_pipeline_master.dir/pipeline/src/util/opencv_fill_poly/line_iterator_cv.cpp.o
[ 50%] Building CXX object cpm-bin/github_bioroboticslab_pipeline_master/bin/CMakeFiles/github_bioroboticslab_pipeline_master.dir/pipeline/src/util/Quaternions.cpp.o
[ 51%] Building CXX object cpm-bin/github_bioroboticslab_pipeline_master/bin/CMakeFiles/github_bioroboticslab_pipeline_master.dir/pipeline/src/util/opencv_fill_poly/line_iterator.cpp.o
[ 52%] Building CXX object cpm-bin/github_bioroboticslab_pipeline_master/bin/CMakeFiles/github_bioroboticslab_pipeline_master.dir/pipeline/src/util/Util.cpp.o
<command-line>:0:0: error: "CPM_CMAKECONFIG_NS" redefined [-Werror]
<command-line>:0:0: note: this is the location of the previous definition
<command-line>:0:0: error: "CPM_CMAKECONFIG_NS" redefined [-Werror]
<command-line>:0:0: note: this is the location of the previous definition
[ 53%] Building CXX object cpm-bin/github_bioroboticslab_pipeline_master/bin/CMakeFiles/github_bioroboticslab_pipeline_master.dir/pipeline/src/util/opencv_fill_poly/poly_line_vertical_iterator.cpp.o
<command-line>:0:0: error: "CPM_CMAKECONFIG_NS" redefined [-Werror]
[ 54%] Building CXX object cpm-bin/github_bioroboticslab_pipeline_master/bin/CMakeFiles/github_bioroboticslab_pipeline_master.dir/pipeline/src/util/opencv_fill_poly/fill_convex_poly_tests.cpp.o
<command-line>:0:0: note: this is the location of the previous definition
<command-line>:0:0: error: "CPM_CMAKECONFIG_NS" redefined [-Werror]
```

And so on and so forth. 
In the file: cpm-bin/github_nebw_mxnetpredict_master/bin/CMakeFiles/github_nebw_mxnetpredict_master.dir/DependInfo.cmake
This got me suspicious:

```
# Preprocessor definitions for this target.
set(CMAKE_TARGET_DEFINITIONS_CXX
  "CPM_CATCH_NS=github_bcachet_cpmcatch_origin_master"
  "CPM_CMAKECONFIG_NS=CPM_TESTING_UPPER_LEVEL_NAMESPACE"
  "CPM_CMAKECONFIG_NS=github_bioroboticslab_cmakeconfig_master"
  "CPM_CPM_CATCH_NS=github_bcachet_cpmcatch_origin_master"
  "CPM_CPM_MXNET_NS=github_nebw_cpm_mxnet_master"
  "CPM_MXNETPREDICT_NS=github_nebw_mxnetpredict_master"
  )
```

**Comment by hmoenck** [comment]
This seems like an issue of cmake 3.5.2. Fixed it by using 3.6.1.
Also watch out to not use conflicting pipeline versions in dependencies. Here: pipeline master vs pipeline mxnet invoked from different deps.
